lib/IRGen/IRGenSIL.cpp

```diff
@@ -3545,15 +3545,31 @@ void IRGenSILFunction::visitTryApplyInst(swift::TryApplyInst *i) {
 }
 
 void IRGenSILFunction::visitFullApplySite(FullApplySite site) {
-  const LoweredValue &calleeLV = getLoweredValue(site.getCallee());
-  
   auto origCalleeType = site.getOrigCalleeType();
   auto substCalleeType = site.getSubstCalleeType();
 
   // SWIFT_ENABLE_TENSORFLOW
-  assert(!origCalleeType->isDifferentiable() && "Differentiable functions "
-         "should not reach here");
-  
+  Optional<LoweredValue> tmpCalleeLV;
+  if (site.getOrigCalleeType()->isDifferentiable()) {
+    auto adFnExp = getLoweredExplosion(site.getCallee());
+    Explosion e;
+    unsigned fieldSize = 1;
+    if (origCalleeType->getRepresentation() ==
+        SILFunctionTypeRepresentation::Thick) {
```

Comment:
I'm not sure this is robust enough. How about other calling conventions such as `@convention(block)`?